andela/hc-wits-kla
# #158175313 Nag User for unresolved jobs

#### What does the PR do?
* Issue nags to the user until he resolves the issues

#### Description of Task to be completed?
* When a User's job goes down, the user is sent a notification email informing him of the same. The system keeps nagging the user every so often until they resolve the job that has gone down

#### How should this be manually tested?
* Clone the branch, git clone https://github.com/andela/hc-wits-kla/tree/ft-nag-user-158175313
* In your terminal, cd hc-wits-kla
* In the project directory, activate the environment, source .env
* Install requirements pip install requirements.txt
* Make migrations python manage.py makemigrations
* Run migrations python manage.py migrate
* Serve the application python manage.py runserver
* In the browser, visit, 127.0.0.1:8000/checks/
* In your terminal, open a new tab and listen for alerts by issuing python manage.py sendalerts
* For real email notifications, provide your email configurations in the settings
* In your browser, create a check(s), set Period to 1 minutes and Grace Time to 1 minute, and the nag interval to a few minutes, when the check goes down and a user is notified, if specified nag interval pass without resolving the check, the system nags the user every after specified minutes until they ping the check. Checks that are on nag mode will show a speak icon beside the nag interval time.
* Go to the console and wait for the nags

#### What are the relevant pivotal tracker stories?
- #158175313

#### Screenshots (if appropriate)
Checks on Nag mode

<img width="1282" alt="screen shot 2018-07-06 at 10 27 07" src="https://user-images.githubusercontent.com/39955231/42366433-d6595a2e-8109-11e8-8041-09f8f8e7cc05.png">

<img width="1249" alt="screen shot 2018-07-07 at 18 17 26" src="https://user-images.githubusercontent.com/39955231/42412578-8ff1db10-8217-11e8-9eee-085539d5244f.png">
